packages/data-stores/src/reader/helpers/index.ts

```diff
@@ -83,23 +83,48 @@ const applyCallbackToPages = < K extends string, T >(
 	};
 };
 
+function getQueryParamFromUrl( param: string ): string | null {
+	let urlObj;
+	try {
+		urlObj = new URL( window.location.href );
```

Comment:
Instead of accessing the `window` inside the `data-store`, you could consider adding the `email_id` as an optional param on the unsubscribe mutation. What do you think?

Reply:
I like it! I'll implement it 👍